warehouse/query-core/src/test/java/datawave/query/util/DateTypeDescriptionIteratorTest.java

```diff
@@ -0,0 +1,198 @@
+
+        withExpectedNextSeekCounts(2, 1);
+        withExpectedFields("FIELD_A");
+        withExpectedBeginEndDates("20250402", "20250404");
+        withDatatypeFilter("datatype-b");
+        drive("20250404", "20250404", "LOADED");
+    }
+
+    private void drive(String start, String stop, String type) throws IOException {
+        Range range = createRangeForDate(start, stop);
+        SortedKeyValueIterator<Key,Value> source = new SortedMapIterator(data);
+
+        Collection<ByteSequence> columnFamily = Collections.singleton(new ArrayByteSequence(type.getBytes()));
+
+        StatDateTypeDescriptionIterator iter = new StatDateTypeDescriptionIterator();
+        iter.init(source, options, env);
+        iter.seek(range, columnFamily, true);
+
+        int count = 0;
+        while (iter.hasTop()) {
+            count++;
+
+            if (log.isTraceEnabled()) {
+                log.trace("tk: {}", iter.getTopKey());
+            }
+
+            Value value = iter.getTopValue();
+            description = DateTypeDescription.deserializeFromString(new String(value.get()));
+
+            iter.next();
+        }
+
+        assertEquals(1, count);
+        assertEquals(expectedNextCount, iter.getNextCount(), "expected next count");
+        assertEquals(expectedSeekCount, iter.getSeekCount(), "expected seek count");
+
+        assertNotNull(description);
+        assertEquals(expectedBeginDate, description.getDateRange()[0]);
+        assertEquals(expectedEndDate, description.getDateRange()[1]);
+        assertEquals(expectedFields, description.getFields());
+
+        // might want to drive an iterator multiple times
+        clearState();
+    }
+
+    private Range createRangeForDate(String start, String stop) {
+        return new Range(start, stop + '~');
+    }
+
+    private void withExpectedNextSeekCounts(int expectedNextCount, int expectedSeekCount) {
+        this.expectedNextCount = expectedNextCount;
+        this.expectedSeekCount = expectedSeekCount;
+    }
+
+    private void withExpectedFields(String... fields) {
+        expectedFields.addAll(List.of(fields));
+    }
+
+    private void withExpectedBeginEndDates(String beginDate, String endDate) {
+        this.expectedBeginDate = beginDate;
+        this.expectedEndDate = endDate;
+    }
+
+    private void withDatatypeFilter(String... types) {
+        options.put(DateIndexIterator.DATATYPE_FILTER, Joiner.on(',').join(types));
+    }
+
+    private void write(String shard, String type, String date, String datatype, String field, int... offsets) {
+        Key key = new Key(shard, type, date + "\u0000" + datatype + "\u0000" + field);
+        BitSet bitset = new BitSet();
+        for (int offset : offsets) {
+            bitset.set(offset);
+        }
+        Value value = new Value(bitset.toByteArray());
+        data.put(key, value);
+    }
+
+    private class StatDateTypeDescriptionIterator extends DateTypeDescriptionIterator {
```

Comment:
This iterator is functionally identical to `DateIndexIteratorTest.StatDateIndexIterator`. I think it's a candidate to be extracted to a top-level class that can then be used by different test classes.